packages/@angular/cli/tasks/npm-install.ts

```diff
@@ -11,22 +12,24 @@ export default Task.extend({
       packageManager = 'npm';
     }
 
-    return new Promise(function(resolve, reject) {
-      ui.writeLine(chalk.green(`Installing packages for tooling via ${packageManager}.`));
-      let installCommand = `${packageManager} install`;
-      if (packageManager === 'npm') {
-        installCommand = `${packageManager} --quiet install`;
-      }
-      exec(installCommand,
-        (err: NodeJS.ErrnoException, _stdout: string, stderr: string) => {
-        if (err) {
-          ui.writeLine(stderr);
-          ui.writeLine(chalk.red('Package install failed, see above.'));
-          reject();
-        } else {
-          ui.writeLine(chalk.green(`Installed packages for tooling via ${packageManager}.`));
-          resolve();
+    return checkYarnOrCNPM().then(function () {
+      return new Promise(function(resolve, reject) {
```

Comment:
There is no need to create a new Promise here, simply returning a value or throwing will work because you're already in a promise chain.